torch/quantization/_adaround.py

```diff
@@ -0,0 +1,260 @@
+import torch
+import torch.nn as nn
+import torch.nn.init as init
+import math
+from torch.quantization.fake_quantize import FakeQuantize
+from torch.quantization.observer import HistogramObserver
+from torch.quantization.qconfig import *
+from torch.quantization.fake_quantize import *
+import torch.nn.qat.modules as nnqat
+from torch.quantization.default_mappings import DEFAULT_QAT_MODULE_MAPPING
+from torch.quantization import QuantStub, DeQuantStub
+import copy
+_supported_modules = {nn.Conv2d, nn.Linear}
+
+# Hyper parameters for loss function
+beta_high = 8
+beta_low = 2
+norm_scaling = 10
+regularization_scaling = .1
+
+# Adaround training parameters
+number_of_epochs = 10
+number_of_calibration_batches = 10
+learning_rate = .1
+
+def computeSqnr(x, y):
+    Ps = torch.norm(x)
+    Pn = torch.norm(x - y)
+    return 20 * torch.log10(Ps / Pn)
+
+def get_module(model, name):
+    ''' Given name of submodule, this function grabs the submodule from given model
+    '''
+    return dict(model.named_modules())[name]
+
+def parent_child_names(name):
+    '''Splits full name of submodule into parent submodule's full name and submodule's name
+    '''
+    split_name = name.rsplit('.', 1)
+    if len(split_name) == 1:
+        return '', split_name[0]
+    else:
+        return split_name[0], split_name[1]
+
+class adaround(FakeQuantize):
```

Comment:
nit: AdaRound for class name

Reply:
maybe `AdaRoundFakeQuantize` to make it more clear that this is a fake quantize module

Reply:
also this should probably be in a different file